cmd/talosctl/cmd/talos/upgrade.go

```diff
@@ -147,6 +147,8 @@ func upgradeGetActorID(ctx context.Context, c *client.Client, opts []client.Upgr
 
 func init() {
 	rebootModes := maps.KeysFunc(machine.UpgradeRequest_RebootMode_value, strings.ToLower)
+	sort.Strings(rebootModes)
```

Comment:
Not sure how it worked before.
But to me it seems to trigger dirty check 50% of time.